### ISSUE TITLE
write_operation delete

### ISSUE THREAD

**Issue by cdugz** [uken/fluent-plugin-elasticsearch#172]
Context: I am setting up streaming replication between 2 ES clusters. On the side of main cluster, I have a client connected to [Changes API](https://github.com/jurgc11/es-change-feed-plugin) plugin, writing all messages to local file

```
{"_index":"sni_cv","_type":"searchdocument","_id":"CRJQ0FF67K490C2PP24Z","_timestamp":"2016-06-07T14:22:55.048Z","_version":2,"_operation":"INDEX","_source":{ <...> }
{"_index":"sni_cv","_type":"searchdocument","_id":"CRJQ0FG68CQ7L1TXWLJP","_timestamp":"2016-06-07T14:22:56.982Z","_version":1,"_operation":"CREATE","_source":{ <...> }
{"_index":"kforce_vac","_type":"searchdocument","_id":"T1BF0G91EHZR","_timestamp":"2016-06-07T14:22:55.762Z","_version":1,"_operation":"DELETE"}
```
Then I pipe all these messages to remote cluster with help of fluentd-plugin-elasticsearch. Couldn't find any mean to ask remote cluster to "delete" a document (last statement in the example).

Any suggestions?

**Comment by pitr** [comment]
I don't understand why you are using this library. `fluent-plugin-elasticsearch` is about sending logs to ElasticSearch, not deleting documents. It feels like you need an ES replication agent instead of log shipping library

**Comment by cdugz** [comment]
>>sending logs to ElasticSearch
this is exactly what I am doing, no? sending messages to ElasticSearch

>>It feels like you need an ES replication agent
does exist? after long googling and asking at IRC, there is no solution except the mentioned plugin. People suggest to use logstash and kafka - but we all know, it's crap

**Comment by cdugz** [comment]
Finally, went with logstash. But anyway, thanks for taking a look

**Comment by cdugz** [comment]
Hi again,

I have used patched version of your plugin (custom behavior on `write_operation`) to achieve my goal. And it works! Our ES cluster survived simulation of DC-outage ;) I have described whole procedure here: https://github.com/cdugz/es-replication

Probably I will submit some PRs to formalize my custom behavior with exotic tags (`write_operation_key`, `write_operation_key_ignore_case`? will see)

Thanks for your support anyways
